jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java

```diff
@@ -422,22 +414,28 @@ protected ChannelEndPoint newEndPoint(SocketChannel channel, ManagedSelector sel
     }
 
     /**
-     * @return the linger time
-     * @see Socket#getSoLinger()
+     * Returns the socket close linger time.
+     *
+     * @return -1 as the socket close linger time is always disabled.
+     * @see java.net.StandardSocketOptions#SO_LINGER
+     * @deprecated don't use as socket close linger time has undefined behavior for non-blocking sockets
      */
-    @ManagedAttribute("TCP/IP solinger time or -1 to disable")
+    @ManagedAttribute(value = "TCP/IP socket close linger time or -1 if disabled", readonly = true)
+    @Deprecated
     public int getSoLingerTime()
     {
-        return _lingerTime;
+        return -1;
     }
 
     /**
-     * @param lingerTime the linger time. Use -1 to disable.
-     * @see Socket#setSoLinger(boolean, int)
+     * @param lingerTime the socket close linger time; use -1 to disable.
+     * @see java.net.StandardSocketOptions#SO_LINGER
+     * @deprecated don't use as socket close linger time has undefined behavior for non-blocking sockets
      */
+    @Deprecated
     public void setSoLingerTime(int lingerTime)
     {
-        _lingerTime = lingerTime;
+        LOG.warn("Ignoring socket close linger time, it has undefined behavior for non-blocking socket");
```

Comment:
I prefer no explanation in warnings.  Just say "Ignoring deprecated soLingerTime" . The full explanation can then be in the javadoc.

If we explain too much in the messages, this sets a bad precedent that cannot be followed for situations where the explanation is longer.